hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/catalog/HoodieHiveCatalog.java

```diff
@@ -510,6 +511,11 @@ private void initTableIfNotExists(ObjectPath tablePath, CatalogTable catalogTabl
     }
 
     flinkConf.setString(FlinkOptions.TABLE_NAME, tablePath.getObjectName());
+
+    List<String> fields = new ArrayList<>();
+    catalogTable.getUnresolvedSchema().getColumns().forEach(column -> fields.add(column.getName()));
+    StreamerUtil.checkPreCombineKey(flinkConf, fields);
```

Comment:
`catalogTable.getUnresolvedSchema().getColumns().collect(Collectos.toList)` ?

Reply:
hi, I need the field name, but getColumns returns the UnresolvedColumn set